service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -171,6 +180,40 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
     // Fetch existing method:
     MethodVersion methodVersion = methodVersionDao.getMethodVersion(request.getMethodVersionId());
 
+    // convert method url to raw GitHub url and use that while calling Cromwell's submit workflow
+    // endpoint
+    String rawMethodUrl;
+    try {
+      rawMethodUrl =
+          MethodUtil.convertToRawGithubUrl(
+              methodVersion.url(),
+              Objects.requireNonNull(
+                  PostMethodRequest.MethodSourceEnum.fromValue(
+                      methodVersion.method().methodSource())),
+              methodVersion.name(),
+              dockstoreService);
+
+      // this could happen if there was no url or empty url received in the Dockstore workflow's
+      // descriptor response
+      if (rawMethodUrl == null || rawMethodUrl.isEmpty()) {
+        String errorMsg =
+            "Error while retrieving WDL url for Dockstore workflow. No workflow url found specified path.";
+        log.error(errorMsg);
+        return new ResponseEntity<>(
+            new RunSetStateResponse().errors(errorMsg), HttpStatus.BAD_REQUEST);
```

Comment:
If this does happen, do we really want to fail the request, or should we instead consider creating a runset which is immediately in the `Failed` state (so that we keep the attempt and the error in the run history?)

Reply:
Hmm that's a good question. I think if it is a bad request / user error (which it is in this case) we should not create a RunSet in failed state. The request was not valid and hence we should only fail the request. We are logging the error from the request so there is a trail for the request. What do you think?

Reply:
Two thoughts:

1. From a user's point of view, I don't know how visible logs are going to be.
2. When we switch from "do everything synchronously in the user request" to "store in the DB and process later", we'll almost certainly have records of things like this against failed runsets. But I don't know that we need to be too worried about how we handle it right now. As long as the user can see the error message easily when they need to, I think it's ok.